### PR TITLE
adds hash and initalize tasks, fixes task series

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -47,13 +45,11 @@ const main = () => {
   }
   const task = argv._.length === 0 ? 'default' : argv._;
   log(`Running ${task}...`);
-  //create dist directory
-  mkdirp.sync(conf.dist);
   loadTasks(conf, (err, runner) => {
     if (err) {
       throw err;
     }
-    runner.run(task);
```
Comment (orthagonal) [inline app.js]: Fixing this seems to make things run in the right order.

File: conf/default.yaml [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+minify: true
 tasks:
   default:
-    -
-      - 'eslint'
-      - 'stylesheets'
-      - 'scripts'
+    - 'initialize'
+    - 'eslint'
+    - 'stylesheets'
```
Comment (jgallen23) [inline conf/default.yaml]: The middle 3 should be an array so they are parallel

File: tasks/hash.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+const async = require('async');
+const ClientKitTask = require('clientkit-task');
+const glob = require('glob');
+const hasher = require('../lib/urlHashes.js');
```
Comment (jgallen23) [inline tasks/hash.js]: Can we pull this logic into this file?

File: tasks/hash.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  process(input, filename, processDone) {
+    async.auto({
+      filesToHash: (done) => {
+        // get the list of matching files
+        glob(input, {}, done);
+      },
+      hashNames: ['filesToHash', (results, done) => {
+        const hashMap = {};
+        results.filesToHash.forEach((file) => {
+          fs.renameSync(file, hasher.hash(file, fs.readFileSync(file)));
```
Comment (jgallen23) [inline tasks/hash.js]: We should never do sync calls.

File: tasks/initialize.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+const ClientKitTask = require('clientkit-task');
+const mkdirp = require('mkdirp');
+const rmdir = require('rmdir');
+const fs = require('fs');
+
+class InitTask extends ClientKitTask {
+  process(distDir, whatever, processDone) {
+    async.auto({
+      clearDist: (done) => {
+        if (fs.existsSync(distDir)) {
```
Comment (jgallen23) [inline tasks/initialize.js]: Exists should be a separate call that isn't sync

**Pull request merged**